contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,293 @@
+				flowInf.Status = FlowInfo_TWO_ENTRIES
+			} else {
+				// got duplicate from same flow; need to indicate reject
+				// or accumulate the stats until next time
+				flowInf.Metric1 = *f.Metric
+				flowInf.Saw1 = true
+				flowInf.Flow1 = f
+				continue
+			}
+		} else { // normal case
+			if flowInf.UUID1 == f.UUID {
+				flowInf.Metric1 = *f.Metric
+				flowInf.Flow1 = f
+				flowInf.Saw1 = true
+			} else {
+				flowInf.Metric2 = *f.Metric
+				flowInf.Flow2 = f
+				flowInf.Saw2 = true
+			}
+		}
+	}
+}
+
+// combineFlows - helper function to combine 2 seen flows into one, based on common minimums
+func combineFlows(f *accRejFlowInfo) *flow.Flow {
+	// perform shallow copy of flow and then update some fields
+	newF := *f.Flow1
+	newMetricReported := *f.Flow1.Metric
+	// take byte and packet count as minimum of the 2 reports
+	if f.Flow1.Metric.ABPackets > f.Flow2.Metric.ABPackets {
+		newMetricReported.ABPackets = f.Flow2.Metric.ABPackets
+	}
+	if f.Flow1.Metric.ABBytes > f.Flow2.Metric.ABBytes {
+		newMetricReported.ABBytes = f.Flow2.Metric.ABBytes
+	}
+	if f.Flow1.Metric.BAPackets > f.Flow2.Metric.BAPackets {
+		newMetricReported.BAPackets = f.Flow2.Metric.BAPackets
+	}
+	if f.Flow1.Metric.BABytes > f.Flow2.Metric.BABytes {
+		newMetricReported.BABytes = f.Flow2.Metric.BABytes
+	}
+	// Note: Metric.Start should never change
+	if f.Flow1.Metric.Last > f.Flow2.Metric.Last {
+		newMetricReported.Last = f.Flow2.Metric.Last
+	}
+	// compute LastUpdateMetric based on previous MetricReported and current newMetricReported
+	var newLastUpdateMetric flow.FlowMetric
+	if f.Status == FlowInfo_OLD_FLOW {
+		newLastUpdateMetric.ABPackets = newMetricReported.ABPackets - f.MetricReported.ABPackets
+		newLastUpdateMetric.ABBytes = newMetricReported.ABBytes - f.MetricReported.ABBytes
+		newLastUpdateMetric.BAPackets = newMetricReported.BAPackets - f.MetricReported.BAPackets
+		newLastUpdateMetric.BABytes = newMetricReported.BABytes - f.MetricReported.BABytes
+		newLastUpdateMetric.Start = f.MetricReported.Last
+		newLastUpdateMetric.Last = newMetricReported.Last
+	} else {
+		// this is the first report for the flow
+		newLastUpdateMetric = *f.Flow1.LastUpdateMetric
+		if f.Flow1.LastUpdateMetric.ABPackets > f.Flow2.LastUpdateMetric.ABPackets {
+			newLastUpdateMetric.ABPackets = f.Flow2.LastUpdateMetric.ABPackets
+		}
+		if f.Flow1.LastUpdateMetric.ABBytes > f.Flow2.LastUpdateMetric.ABBytes {
+			newLastUpdateMetric.ABBytes = f.Flow2.LastUpdateMetric.ABBytes
+		}
+		if f.Flow1.LastUpdateMetric.BAPackets > f.Flow2.LastUpdateMetric.BAPackets {
+			newLastUpdateMetric.BAPackets = f.Flow2.LastUpdateMetric.BAPackets
+		}
+		if f.Flow1.LastUpdateMetric.BABytes > f.Flow2.LastUpdateMetric.BABytes {
+			newLastUpdateMetric.BABytes = f.Flow2.LastUpdateMetric.BABytes
+		}
+		newLastUpdateMetric.Last = newMetricReported.Last
+	}
+	newF.Metric = &newMetricReported
+	newF.LastUpdateMetric = &newLastUpdateMetric
+
+	// FIXME - What should we do with UpdateCount?
```

Comment:
We believe that `UpdateCount` is not used by Security Advisor at all.

Reply:
In the meantime we'll just take the value provided by Flow1.